bob/src/hw_metrics_collector.rs

```diff
@@ -242,52 +266,83 @@ impl DiskStatCollector {
         }
 
         DiskStatCollector {
-            iostat_avl: true,
+            procfs_avl: true,
             disk_metric_data,
         }
     }
 
     fn collect_and_send_metrics(&mut self) -> Result<(), CommandError> {
-        let iostat_lines = self.collect_iostat()?;
-        for i in iostat_lines {
+        let uptime = self.uptime()?;
+        let diskstats = self.diskstats()?;
+        for i in diskstats {
             let lsp: Vec<&str> = i.split_whitespace().collect();
-            if lsp.len() == 21 {
-                if let Some(metric_prefix) = self.disk_metric_data.get(lsp[0]) {
-                    if let (Ok(rs), Ok(ws)) = (
-                        // readops per s count is in 1st column
-                        Self::parse_iostat_result(lsp[1]),
-                        // writeops per s count is in 7th column
-                        Self::parse_iostat_result(lsp[7]),
+            if lsp.len() >= 8 {
+                if let Some(metric_prefix) = self.disk_metric_data.get(lsp[2]) {
+                    if let (Ok(r_ios), Ok(w_ios)) = (
+                        // successfull reads count is in 3rd column
+                        lsp[3].parse::<f64>(),
+                        // successfull writes count is in 7th column
+                        lsp[7].parse::<f64>(),
                     ) {
-                        let iops = rs + ws;
+                        let iops = (r_ios + w_ios) / uptime;
 
                         let gauge_name = format!("{}_iops", metric_prefix);
                         gauge!(gauge_name, iops);
-                    } else {
-                        return Err(CommandError::Primary("Can't parse iostat output into float".to_string()));
-                    }
-
-                    if let (Ok(rwait), Ok(wwait)) = (
-                        // readops wait time is in 5th column
-                        Self::parse_iostat_result(lsp[5]),
-                        // writeops wait time is in 11th column
-                        Self::parse_iostat_result(lsp[11]),
-                    ) {
-                        let iowait = rwait + wwait;
 
-                        let gauge_name = format!("{}_iowait", metric_prefix);
-                        gauge!(gauge_name, iowait);
+                        if lsp.len() >= 11 {
+                            if let (Ok(w_ticks), Ok(r_ticks)) = (
+                                // time spend on write is in 10th column
+                                lsp[10].parse::<f64>(),
+                                // time spend on read is in 6th column
+                                lsp[6].parse::<f64>(),
+                            ) {
+                                let iowait = r_ticks / r_ios + w_ticks / w_ios;
```

Comment:
Same problem here. Also, you should check that `r_ios` and `w_ios` is greater than 0 (otherwise you will get a NaN)